pkg/trace/agent/agent.go

```diff
@@ -287,17 +287,19 @@ func (a *Agent) sample(pt ProcessedTrace) (sampled bool, rate float64) {
 	var sampledPriority, sampledScore bool
 	var ratePriority, rateScore float64
 
-	if _, ok := pt.GetSamplingPriority(); ok {
-		sampledPriority, ratePriority = a.PrioritySampler.Add(pt)
-	}
-
 	if traceContainsError(pt.Trace) {
 		sampledScore, rateScore = a.ErrorsScoreSampler.Add(pt)
 	} else {
 		sampledScore, rateScore = a.ScoreSampler.Add(pt)
 	}
 
-	return sampledScore || sampledPriority, sampler.CombineRates(ratePriority, rateScore)
+	// If sampling priority is set, it takes precedence over sampledScore
```

Comment:
I’ve generally tried to introduce the convention that comments which describe what the body of an “if” clause is doing would be placed on the first line below the “if” statement. Would you mind either removing this comment or doing that?